### PR TITLE
701 make reported errors more useful

### PR DESCRIPTION
**Pull request by leite08**
Ref. metriport/metriport-internal#701

### Dependencies

none

### Description

Do some processing while dealing with errors mid FHIR processing, so we're better equipped to deal with them.

Both from FHIR server and FHIR converter.

### Release Plan

- nothing special

### CODE REVIEW

File: api/app/src/external/commonwell/document/document-query.ts [hunk 1]
```diff
@@ -203,6 +206,63 @@ async function internalGetDocuments({
...
+  log,
+}: {
+  errors: OperationOutcome[];
+  context: Record<string, unknown>;
+  log: ReturnType<typeof Util.out>["log"];
+}): void {
+  const errorsByCategory = groupCWErrors(errors);
+  for (const [category, errors] of Object.entries(errorsByCategory)) {
+    const msg = `Document query error - ${category}`;
+    log(`${msg}: ${JSON.stringify(errors)}`);
```
Comment (Goncharo) [inline api/app/src/external/commonwell/document/document-query.ts:221]: nit: `stringify(errors, null, 2)` for nicer formatting?

File: api/app/src/external/commonwell/error-categories.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      | ErrorCategory
+      | undefined;
+    if (category) addToCategory(category, i);
+    else addToCategory("Metriport could not determine", i);
+  });
+  return errorsByCategory;
+}
+
+// Keep this sorted from the most specific to the most generic
+export const errorCategories = [
```
Comment (Goncharo) [inline api/app/src/external/commonwell/error-categories.ts:29]: 🧙

**Review by Goncharo — APPROVED** [review]
looks great, :shipit:  - one small nit

**Review by Orta21 — APPROVED** [review]

File: api/app/src/external/commonwell/document/document-query.ts (hunk 1 above)
Comment (leite08) [reply api/app/src/external/commonwell/document/document-query.ts:221]: avoiding these until we [fix multiline logs on CloudWatch](https://github.com/metriport/metriport-internal/issues/693) - currently these multiline logs make it much harder to read the logs

**Pull request merged**